pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -252,6 +255,106 @@ pub(super) async fn download_index_part(
     Ok(index_part)
 }
 
+/// index_part.json objects are suffixed with a generation number, so we cannot
+/// directly GET the latest index part without doing some probing.
+///
+/// In this function we probe for the most recent index in a generation <= our current generation.
+/// See "Finding the remote indices for timelines" in docs/rfcs/025-generation-numbers.md
+pub(super) async fn download_index_part(
+    storage: &GenericRemoteStorage,
+    tenant_id: &TenantId,
+    timeline_id: &TimelineId,
+    my_generation: Generation,
+) -> Result<IndexPart, DownloadError> {
+    if my_generation.is_none() {
+        // Operating without generations: just fetch the generation-less path
+        return do_download_index_part(storage, tenant_id, timeline_id, my_generation).await;
+    }
+
+    // Stale case: If we were intentionally attached in a stale generation, there may already be a remote
+    // index in our generation.
+    //
+    // This is an optimization to avoid doing the listing for the general case below.
+    let r_current = do_download_index_part(storage, tenant_id, timeline_id, my_generation).await;
```

Comment:
```suggestion
    let res = do_download_index_part(storage, tenant_id, timeline_id, my_generation).await;
```

This could be just res, similarly in following steps as well? There's no usage except launder the notfound or return the result?

Reply:
It could.  The intent of these variable names is to help readability -- it makes it more obvious that this line is downloading the current generation's index, the later one is downloading the current generation, etc.

Then again there are also comments here explaining what each request is doing, and avoiding the unique names will avoid giving the impression that we're going to use all the results later.

Reply:
These are changed to `res` in 008545c5b10f7118fffecdffa51126263a32c7f3